filters/encrypt/filter.go

```diff
@@ -0,0 +1,554 @@
+	// rotated with an event that has a payload satisfying the RotateWrapper
+	// interface. If an event's payload satisfies the EventWrapperInfo
+	// interface, event specific HmacSalt will be used for operations on that
+	// specific event.
+	HmacSalt []byte
+
+	// Info for deriving a hmac-sha256 operations key (can be nil). This may be
+	// rotated with an event that has a payload satisfying the RotateWrapper
+	// interface.  If an event's payload satisfies the
+	// EventWrapperInfo interface, event specific HmacInfo will be used for
+	// operations on that specific event.
+	HmacInfo []byte
+
+	// FilterOperationOverrides provide an optional a set of runtime overrides
+	// for the FilterOperations to be applied to DataClassifications.
+	//
+	// Normally, the filter operation applied to a field is determined by the
+	// operation specified in it's "classified" tag. If no operation is
+	// specified in the tag, then a set of reasonable default filter operations
+	// are applied.
+	//
+	// FilterOperationOverrides provides the ability to override an event's
+	// "classified" tag settings.
+	FilterOperationOverrides map[DataClassification]FilterOperation
+
+	l sync.RWMutex
+}
+
+// Reopen is a no op for Filters.
+func (af *Filter) Reopen() error {
+	return nil
+}
+
+// Type describes the type of the node as a Filter.
+func (ef *Filter) Type() eventlogger.NodeType {
+	return eventlogger.NodeTypeFilter
+}
+
+// Rotate supports rotating the filter's wrapper, salt and info via the options:
+// WithWrapper, WithSalt, WithInfo
+func (ef *Filter) Rotate(opt ...Option) {
+	opts := getOpts(opt...)
+	ef.l.Lock()
+	defer ef.l.Unlock()
+	if opts.withWrapper != nil {
+		ef.Wrapper = opts.withWrapper
+	}
+	if opts.withSalt != nil {
+		ef.HmacSalt = opts.withSalt
+	}
+	if opts.withInfo != nil {
+		ef.HmacInfo = opts.withInfo
+	}
+}
+
+// Process will encrypt or hmac-sha256 string and []byte fields which are tagged
+// as SensitiveClassification.  Fields that are tagged SecretClassification will
+// be redacted.
+//
+// If the event payload satisfies the WrapperPayload interface, then the
+// payload's Wrapper(), HmacSalt() and HmacInfo() will be used to rotate the
+// filter's wrappers for ongoing filtering operations.  Events matching this
+// WrapperPayload interface are not sent along in the pipeline and a nil with no
+// errors is immediately returned after the wrapper has been rotated.
+//
+// If the event payload satisfies the EventWrapperInfo interface, then the
+// payload's EventId(), HmacSalt() and HmacInfo() will be used to for filtering
+// operations for just the single event being processed.
+func (ef *Filter) Process(ctx context.Context, e *eventlogger.Event) (*eventlogger.Event, error) {
+	const op = "event.(Filter).Process"
+	if e == nil {
+		return nil, fmt.Errorf("%s: missing event: %w", op, ErrInvalidParameter)
+	}
+	if e.Payload == nil {
+		return e, nil
+	}
+
+	if i, ok := e.Payload.(RotateWrapper); ok {
```

Comment:
Optional: You could combine this and the below `if` clause into a single switch. Perhaps with the rotate case calling out to a simple function for brevity if cleaner. Or, you can leave it as is. I don't have a strong preference, just noted the similarity of the checks.

Reply:
I think, I'll leave as is for now... I'm not sure why, but I like the two ifs since once returns and the other falls through to the rest of the function.